src/tests/basic/hand.ts

```diff
@@ -0,0 +1,162 @@
+import { GameState } from "../../framework/types";
+import { P1, initDummyGameState } from "../testhelper";
+import {
+  discardAll,
+  discardCard,
+  drawCard,
+} from "../../framework/actions/hand";
+import { expect } from "chai";
+
+describe("basic", () => {
+  describe("hand", () => {
+    let $: GameState;
+    beforeEach(() => {
+      $ = initDummyGameState();
+      $.players[P1].hand = [ "Nautical Dog", "Mad Man", "Bombaster" ];
+      $.players[P1].deck = [
+        "Careless Musketeer",
+        "Bloodrage Ogre",
+        "Makeshift Rambaster",
+      ];
+    });
+
+    describe("discardAll()", () => {
+      it("should discard all of the player's hand into the discard", () => {
+        discardAll($);
+
+        expect($.players[P1].hand.length).to.equal(0);
+        expect($.players[P1].discard.length).to.equal(3);
+      });
+
+      it("should not get rid of existing discarded cards", () => {
+        $.players[P1].discard = [ "Nautical Dog" ];
+
+        discardAll($);
+
+        expect($.players[P1].hand.length).to.equal(0);
+        expect($.players[P1].discard.length).to.equal(4);
+      });
+
+      it("should not fail if the player's hand is empty", () => {
+        $.players[P1].hand = [];
+
+        discardAll($);
+
+        expect($.players[P1].hand.length).to.equal(0);
+        expect($.players[P1].discard.length).to.equal(0);
+      });
+    });
+
+    describe("discardCard()", () => {
+      it("moves cards from hand to discard pile when discarding", () => {
+        discardCard($, "Nautical Dog");
+
+        expect($.players[P1].hand.length).to.equal(2);
+        expect($.players[P1].hand.indexOf("Nautical Dog")).to.equal(-1);
```

Comment:
```suggestion
        expect($.players[P1].hand.includes("Nautical Dog")).to.equal(false);
```

We typically prefer `includes` over `indexOf` in cases where we don't care about index